src/decode.h

```diff
@@ -250,7 +264,57 @@ static inline uint32_t sign_extend_b(const uint32_t x)
     return (int32_t) ((int8_t) x);
 }
 
+/* allocate a new basic block */
+static inline bool block_init(block_t *block, uint32_t bit)
+{
+    assert(block);
+
+    block->capacity = 1 << bit;
+    block->ir = malloc(sizeof(struct rv_insn_t) * block->capacity);
+    return (bool) (uintptr_t) block->ir;
+}
+
+/* free the basic block */
+static inline void block_free(block_t *block)
+{
+    assert(block);
+    free(block->ir);
+}
+
+/* check the queue is full or not */
+static inline bool block_is_full(block_t *block)
+{
+    assert(block);
+    return (block->front - block->rear) == block->capacity;
+}
+
+/* check the queue is empty or not */
+static inline bool block_is_empty(block_t *block)
+{
+    assert(block);
+    return block->front == block->rear;
+}
+
+/* get the current memory block in queue */
+static inline struct rv_insn_t *queue_current(block_t *block)
```

Comment:
Apply the consistent naming scheme. That is, we shall have the helper function below:
* `block_is_full`, `block_is_full`
* `block_current_insn`
* `block_enqueue`
* `block_dequeue`